src/GovUk.Education.ExploreEducationStatistics.Publisher.Tests/Services/NotificationsServiceTests.cs

```diff
@@ -0,0 +1,222 @@
+            }
+
+            var storageQueueService = new Mock<IStorageQueueService>(MockBehavior.Strict);
+            storageQueueService.Setup(mock => mock.AddMessages(
+                PublicationQueue,
+                It.IsAny<List<PublicationNotificationMessage>>()))
+                .Returns(Task.CompletedTask);
+
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                var notificationsService = BuildNotificationsService(
+                    contentDbContext, storageQueueService.Object);
+
+                await notificationsService.NotifySubscribersIfApplicable(release1.Id, release2.Id, release3.Id);
+
+                storageQueueService.Verify(mock => mock.AddMessages(
+                    PublicationQueue,
+                It.Is<List<PublicationNotificationMessage>>(list =>
+                    list.SequenceEqual(new List<PublicationNotificationMessage>
+                    {
+                        new()
+                        {
+                            Name = release1.Publication.Title,
+                            PublicationId = release1.Publication.Id,
+                            Slug = release1.Publication.Slug,
+                        },
+                        new()
+                        {
+                            Name = release3.Publication.Title,
+                            PublicationId = release3.Publication.Id,
+                            Slug = release3.Publication.Slug,
+                        },
+                    }))), Times.Once);
+            }
+
+            MockUtils.VerifyAllMocks(storageQueueService);
+        }
+
+        [Fact]
+        public async Task NotifySubscribersIfApplicable_NotifyOnSet()
+        {
+            var release = new Release
+            {
+                Id = Guid.NewGuid(),
+                Publication = new Publication
+                {
+                    Id = Guid.NewGuid(),
+                    Title = "pub title",
+                    Slug = "pub-slug",
+                },
+                ReleaseStatuses = new List<ReleaseStatus>
+                {
+                    new()
+                    {
+                        Id = Guid.NewGuid(),
+                        ApprovalStatus = ReleaseApprovalStatus.Approved,
+                        Created = DateTime.UtcNow,
+                        NotifySubscribers = true,
+                        NotifiedOn = DateTime.UtcNow,
+                    },
+                },
+            };
+            var contentDbContextId = Guid.NewGuid().ToString();
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                await contentDbContext.AddRangeAsync(release);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            var storageQueueService = new Mock<IStorageQueueService>(MockBehavior.Strict);
```

Comment:
Similar to my comment below on the `NotifySubscribersIfApplicable_None` you could strip out the `storageQueueService ` from this test. The test will fail if there's an interaction with the strict mock.